### PR TITLE
Fix: allow whitespace in list of extras for a package

### PR DESCRIPTION
**Pull request by finswimmer**
This fix allows having whitespaces in the list of extras during `poetry add`, e.g. `poetry add databases[postgresql, sqlite]`.

# Pull Request Check List

Resolves: https://github.com/python-poetry/poetry/issues/2980

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: tests/console/commands/test_init.py [hunk 1]
```diff
@@ -556,3 +556,13 @@ def test_predefined_and_interactive_dev_dependencies(tester, repo):
     assert expected in output
     assert 'pytest-requests = "^0.2.0"' in output
     assert 'pytest = "^3.6.0"' in output
+
+
+def test_add_package_with_extras_and_whitespace(app):
+    command = app.find("init")
+    result = command._parse_requirements(["databases[postgresql, sqlite]"])
```
Comment (abn) [inline tests/console/commands/test_init.py:563]:
```suggestion
def test_add_package_with_extras_and_whitespace(tester):
    result = tester._parse_requirements(["databases[postgresql, sqlite]"])
```

File: poetry/console/commands/init.py [hunk 2]
```diff
@@ -373,7 +373,7 @@ def _parse_requirements(
         for requirement in requirements:
             requirement = requirement.strip()
             extras = []
-            extras_m = re.search(r"\[([\w\d,-_]+)\]$", requirement)
+            extras_m = re.search(r"\[([\w\d,-_ ]+)\]$", requirement)
```
Comment (abn) [inline poetry/console/commands/init.py:376]: Shoud we just use `\s` here?

**Review by abn — CHANGES_REQUESTED** [review]

File: poetry/console/commands/init.py (hunk 2 above)
Comment (finswimmer) [reply poetry/console/commands/init.py:376]: The range of `\s` is to wide. I don't think supporting e.g. tab characters here.

File: tests/console/commands/test_init.py (hunk 1 above)
Comment (finswimmer) [reply tests/console/commands/test_init.py]: It look like it must be 'test._command._parse_requirements` ...

File: poetry/console/commands/init.py (hunk 2 above)
Comment (abn) [reply poetry/console/commands/init.py:376]: Hmm, I guess for this case it is fine (ish). For PEP 508, whitespace typically refers to "non-line breaking whitespace". That is narrowier than `\s` and but wider than ` `. I doubt anyone is going to type in `foo[a, \t b]` :smirk:

**Review by abn — APPROVED** [review]

**Review by abn — CHANGES_REQUESTED** [review]
Issue talks about `add` command but the fix is in `init` command?

**Review by abn — APPROVED** [review]

**Pull request merged**